### PR TITLE
test (gradle-plugin/it) : Add test for setting ServiceAccount name in generated controller (#2187)

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description
Test case covering #2187 

Failing cases have been commented. I'll remove comment in #1889 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] My code follows the style guidelines of this project
 - [X] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [X] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [X] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [X] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: gradle-plugin/it/src/it/serviceaccountname-via-groovy-dsl/build.gradle [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    resources {
+        serviceAccount = "my-serviceaccount"
+    }
+    enricher {
+        config {
+            'jkube-serviceaccount' {
+                skipCreate = true
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (rohanKanojia) [reply gradle-plugin/it/src/it/serviceaccountname-via-groovy-dsl/build.gradle:34]: ServiceAccountEnricher seems to create additional ServiceAccount resource referenced in Deployment. This flag tells ServiceAccountEnricher to not do so.

https://github.com/eclipse/jkube/blob/0dde32e57a15640a0877348e67ebdf8500b0cad0/jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/ServiceAccountEnricher.java#L65

File: gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ServiceAccountNameViaGroovyDSLIT.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+class ServiceAccountNameViaGroovyDSLIT {
+  @RegisterExtension
+  private final ITGradleRunnerExtension gradleRunner = new ITGradleRunnerExtension();
+
+  static Stream<Arguments> testInput() {
+    return Stream.of(
+//        arguments("no-fragment", new String[] {}),
+//        arguments("fragment", new String[] {"-Pjkube.resourceDir=./regularFragmentDir"}),
```
Comment (manusa) [inline gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ServiceAccountNameViaGroovyDSLIT.java:40]: From the test expectations, the assumption here is that a fragment with an unrelated field has no action over the serviceAccountName field, right?

Could we state this more clearly? A simple "fragment" seems to convey the opposite assumption (I have a fragment with a service account -> what we test in the next case). So maybe an `unrelated-fragment` name might be more suitable.

Comment (rohanKanojia) [reply gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ServiceAccountNameViaGroovyDSLIT.java]: I've renamed`fragment` -> `unrelated-fragment`

File: gradle-plugin/it/src/it/serviceaccountname-via-groovy-dsl/build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (rohanKanojia) [reply gradle-plugin/it/src/it/serviceaccountname-via-groovy-dsl/build.gradle]: I've removed `skipCreate` configuration.

**Review by manusa — APPROVED** [review]

**Pull request merged**